### ISSUE TITLE
Error when multipart request schema includes types other than string or file

### ISSUE THREAD

**Issue by colonel-sanders** [ardatan/feTS#1309]
**Describe the bug**

The `@whatwg-node/fetch` library expects that the value of a multipart form section will be [either a string or a Blob](https://github.com/ardatan/whatwg-node/blob/master/packages/node-fetch/src/FormData.ts).  However, feTS client doesn't enforce this constraint when [building a multipart form object from the submitted `formData`](https://github.com/ardatan/feTS/blob/release-1710384184544/packages/fets/src/client/createClient.ts#L212). If the schema for a multipart request includes a type other than string or file, that value is passed in to the `FormData` object as-is, and an error occurs downstream during the fetch.

**To Reproduce** Steps to reproduce the behavior:

Given an OpenAPI spec (JSON wrapped as TS) with a multipart endpoint:

```
export default {
    openapi: "3.0.3",
    info: {
        title: "Multipart Example",
        description:
            "Demonstration of the error when a multipart field isn't a string or file",
        version: "1.0.0",
    },
    servers: [
        {
            url: "http://0.0.0.0:9999",
        },
    ],
    paths: {
        "/images": {
            post: {
                requestBody: {
                    content: {
                        "multipart/form-data": {
                            schema: {
                                type: "object",
                                properties: {
                                    image: {
                                        type: "string",
                                        format: "binary",
                                    },
                                    licensed: {
                                        type: "boolean",
                                    },
                                },
                            },
                        },
                    },
                },
                responses: {
                    "201": {
                        content: {
                            "application/octet-stream": {
                                schema: {
                                    type: "string",
                                    format: "binary",
                                },
                            },
                        },
                        description: "OK",
                    },
                },
            },
        },
    },
} as const
```

and a client built and invoked like:
```
import { createClient, type NormalizeOAS } from "fets"
import fs from "fs"
import multipartSpec from "./multipart-example-openapi.ts"

type MultipartApi = NormalizeOAS<typeof multipartSpec>
const client = createClient<MultipartApi>({ endpoint: "http://0.0.0.0:9999" })

const image = fs.readFileSync("my-image.png")
const reply = await client["/images"].post({
    formData: {
        image: new File([image], "my-image.png", { type: "image/x-png" }),
        licensed: true,
    },
})
console.log(await reply.text())

```

I would expect a successful multipart submission. Instead, I get the error:

```
value has no function stream - at node_modules/@whatwg-node/node-fetch/cjs/FormData.js:98
```

This occurs because the `getStreamFromFormData` method expects field values to be either strings or Blobs, but the field "licensed" is a boolean.

**Expected behavior**

Any primitive value should come through as a multipart section of Content-Type "text/plain" (the default).

**Environment:**

- OS: macOS 12.7.3
- `fets`: 0.7.1, 0.7.2
- `@whatwg-node/fetch`: 0.9.16
- `@whatwg-node/node-fetch`: 0.5.5
- NodeJS: reproduced on v18.13.0, v20.5.0, v20.11.0

**Additional context**

I'm happy to contribute a patch for this issue. Two potential approaches come to mind, but open to others:

- The simplest approach seems to be to stringify any non-Blob value within [the FormData creation in createClient](https://github.com/ardatan/feTS/blob/release-1710384184544/packages/fets/src/client/createClient.ts#L212). This is what I'm doing as a workaround right now (though implemented slightly differently - I'm passing in a patched `fetchFn`).

- Another potential approach is to handle the issue at the schema level. My read of the [OpenAPI spec](https://swagger.io/specification/#special-considerations-for-multipart-content) is that any (JSON Schema) primitive type is allowable for a field here, but that doesn't mean that feTS has to support it. Two potential approaches here are (1) reject fields in a multipart schema that aren't string or file, or (2) generate the typing of any non-file field as string regardless of the specified type.

**Comment by ardatan** [comment]
Thanks for the detailed report!
I think stringifying the value makes sense :)

**Comment by colonel-sanders** [comment]
Fixed in 0.8.0

**Comment by ardatan** [comment]
Thanks for the report and the PR :)
